### PR TITLE
feat: `signOut()`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

implement `signOut()` with optional callback

## Testing

after click `signOut()`:
- keychain data removed
- the refresh token removed in DB

## Reviewers

@logto-io/eng

### CODE REVIEW

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                        completion?(.failure(error: Errors.SignOut(type: .unableToRevokeToken, innerError: $0)))
+                        return
+                    }
+
+                    completion?(.success)
+                }
+            }
+        }
+
+        accessTokenMap = [:]
```
Comment (simeng-li) [inline Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift:35]: should we run these ahead of the revoke logic?

**Review by simeng-li — APPROVED** [review]
overall LGTM

**Review by wangsijie — APPROVED** [review]
LGTM

**Review by demonzoo — APPROVED** [review]

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift:35]: Sure

Comment (gao-sun) [reply Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift:35]: LOG-1473

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//
+//
+//  Created by Gao Sun on 2022/2/4.
+//
+
+import Foundation
+import Logto
+
+public extension LogtoClient {
+    func signOut(completion: LogtoCompletion<Errors.SignOut>? = nil) {
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply Sources/LogtoClient/LogtoClient/LogtoClient+SignOut.swift:12]: https://linear.app/silverhand/issue/LOG-1474/research-sign-out-on-endsessionendpoint

**Pull request merged**